.github/workflows/daily.yml

```diff
@@ -0,0 +1,129 @@
+name: "Daily Cron"
+
+on:
+  # Run daily at midnight
+  schedule:
+    - cron: "0 0 * * *"
+
+  # Allow workflow to be manually run from the GitHub UI
+  workflow_dispatch:
+
+jobs:
+
+  instrumented-tests:
+    name: "Instrumented Tests"
+    timeout-minutes: 30
+    runs-on: macos-11
+    steps:
+      - name: "Checkout release branch"
+        uses: actions/checkout@v2
+        with:
+          repository: mparticle/mparticle-android-sdk
+          ref: development
+      - name: "Install JDK 11"
+        uses: actions/setup-java@v2
+        with:
+          distribution: "zulu"
+          java-version: "11"
+      - name: "Run Instrumented Tests"
+        uses: reactivecircus/android-emulator-runner@v2.20.0
+        with:
+          api-level: 29
+          script: ./gradlew :android-core:cAT :android-kit-base:cAT --stacktrace
+      - name: "Archive Instrumented Test Results"
+        uses: actions/upload-artifact@v2
+        if: ${{ always() }}
+        with:
+          name: test-results
+          path: android-core/build/reports/androidTests/connected/**
+
+  unit-tests:
+    name: "Unit Tests"
+    timeout-minutes: 15
+    runs-on: ubuntu-18.04
+    steps:
+      - name: "Checkout release branch"
+        uses: actions/checkout@v2
+        with:
+          repository: mparticle/mparticle-android-sdk
+          ref: development
+      - name: "Install JDK 11"
+        uses: actions/setup-java@v2
+        with:
+          distribution: "zulu"
+          java-version: "11"
+      - name: "Run Unit Tests"
+        run: ./gradlew test
+      - name: "Android Test Report"
+        uses: asadmansr/android-test-report-action@v1.2.0
+        if: ${{ always() }}
+
+  update-kits:
+    name: "Update Kits"
+    needs: [instrumented-tests, unit-tests]
+    runs-on: macos-11
```

Comment:
I forgot why we originally chose it, but is it possible to run this on ubuntu?

Reply:
it was missing a dependency when we last tried (just faster to use macOS since we know it works)...we can try again, but let's make it part of another PR if it works